hw/flash_ctrl.rdl

```diff
@@ -0,0 +1,83 @@
+addrmap flash_ctrl {
+      reg {
+        field {
+            sw = rw;
+            onwrite = woclr;
+            desc = "Error-related interrupts";
+        } ERROR[0:0];
+        field {
+            sw = rw;
+            onwrite = woclr;
+            desc = "Event-related interrupts";
+        } EVENT[1:1];
+    } INTERRUPT_STATE @ 0x00;
+    reg {
+        field {
+            sw = rw;
+            desc = "Enable error interrupt";
+        } ERROR[0:0];
+        field {
+            sw = rw;
+            desc = "Enable event interrupt";
+        } EVENT[1:1];
+    } INTERRUPT_ENABLE @ 0x04;
+
+    reg {
+        field {
+            sw = rw;
+            desc = "Page size";
+        } PAGE_SIZE[31:0];
+    } PAGE_SIZE @ 0x08;
+
+    reg {
+        field {
+            sw = rw;
+            desc = "The page number for read, write, erase operations";
+        } PAGE_NUM[31:0];
+    } PAGE_NUM @ 0x0C;
+
+    reg {
+        field {
+            sw = rw;
+            desc = "The page buffer address for read/write operations";
+        } PAGE_ADDR[31: 0];
+    } PAGE_ADDR @ 0x10;
```

Comment:
So, this is like a DMA controller? We program the address to read to / write from, then use `CONTROL` to start the read, write, or erase, then check the status with `OP_STATUS`?

Reply:
Yes, exactly! I had intended to use the FIFO register for data transfer, but that adds extra complexity for this kind of dummy controller.